specs/access-node-architecture/index.md

```diff
@@ -0,0 +1,121 @@
+- The `Blockchain Data Service` encapsulates the live synchronization systems provided through consensus follower and execution state sync. It provides access to that state through the `Protocol API` and the `State Streamer` interfaces.
+- The `State Streamer` interface is a generic component which can allow any consumer, such as the `DPS`, to access the state tracked by the `Blockchain Data Service`.
+
+# Non-goals
+
+- Does not address long term historical access (see additional questions)
+- Does not address operational issues for public node operators (rate-limiting, access control etc)
+
+# Detail level objectives
+
+- The handling of raw protocol and execution state data received from the peer-to-peer network is managed in the `Blockchain Data Service`. This state is provided to callers via the `Protocol API` and streamed to consumers of the `State Streamer`.
+- `Blockchain Data Service` state is accessed using `State Streamer Registration API` and via `State Streamer` clients.
+- `DPS` indexes execution state that has been delivered live from the protocol by `State Streamer`. The indexed state is used to answer script execution requests from the `DPS API`.
+
+# Design
+
+![Architecture Diagram](images/architecture.png)
+
+# In detail
+
+To date the implementation of the Access Node has provided the following: 
+
+- Participates as a *staked* node within the Flow network
+- Participates in the staked gossip network to follow protocol state
+- Handling of requests to the Access API from public callers for current live spork and
+    - Performs basic request validation
+    - Delegates script execution to upstream Execution Nodes
+    - Delegates transactions to Collection Nodes for processing by the network
+
+Script execution delegation to ENs is an obvious low hanging fruit limiting Access Node scalability and impacting transaction throughput capacity. In addition, scaling of Access API request handling is constrained by tight coupling of code within Access Node. The introduction of the DPS took the first step towards queryable locally held state that could be leveraged to scale Access Node. Subsequently, execution state sync provided a decentralized way to share execution state across the network. 
+
+## Revised ownership and responsibility
+
+By managing all chain synchronization concerns `Blockchain Data Service` provides a unified set of functionality that `Access Node` and `Observer Service` are built on, and which can be used as a standalone component for following trusted chain state more generally.
+
+### Blockchain Core service
+
+- Follows protocol consensus and execution state sync
+- Serves the `State Streamer API`
+- Serves the `Protocol API`
+
+The new `Access Node` builds on the original node capabilities above: 
+### Access Node
+
+- Participates as a *staked* node within the Flow network
+- Maintains sync using `Blockchain Data Service`
```

Comment:
This sentence doesn't sound entirely accurate to me. The Access Node doesn't really "use" the Blockchain Data Service, because it *is* the Blockchain Data Service

Reply:
```suggestion
- Stays up to date with protocol state and execution data
```